### DISCUSSION TITLE
Toggleable group active state with JavaScript (similar to group-hover)

### DISCUSSION THREAD

**Discussion by Swennet** [tailwindlabs/tailwindcss#7382] [Ideas]
It would be nice to be able to style a group's children depending on if that group's state is set to active or inactive in JavaScript. This would work very similar to how `group-hover` works. 

For example, a data-attribute named data-active is added with a click event in JavaScript to the group element.

```
<div class="group" /* data-active */>

    <div class="opacity-0 group-data-active:opacity-100">
	Show this when "data-active" is set on the group element.
    </div>

</div>
```

Is this already possible?

If not, it seems like this would add a lot of flexibility to changing styles with a simple JavaScript selector, without specifically adding and removing those styles with JavaScript's classList functionality.

**Comment by alexanderbluhm** [comment]
Hi @Swennet 
This is possible with a custom [Plugin](https://tailwindcss.com/docs/plugins#adding-variants).
https://play.tailwindcss.com/yTikYBoLw8?file=config
```js
plugin(function ({ addVariant }) {
  addVariant('group-data-active', '.group[data-active] &')
}),
```

**Reply by Swennet** [reply]
Works exactly how I imagined! I tried messing around with custom Plugins first but got the syntax wrong. Thanks so much for the help!

**Comment by Swennet** [comment]
To anyone stumbling upon this, I found this doesn't quite work well with nested groups so I changed the plugin to work slightly different to circumvent this issue.

```
plugin(function ({ addVariant }) {
    addVariant('toggle-active', '.toggle[data-active=true] &');
}),
```

Usage works like this:

```
<div class="toggle">
    <div class="text-white toggle-active:text-black">
        This text will turn black on toggle.
    </div>
</div>
```

Now when you add a `data-active="true"` attribute to your `toggle` element the text will turn black.
